projects/core/src/cms/store/cms-state.ts

```diff
@@ -1,17 +1,27 @@
-import { EntityState } from '../../state';
+import { EntityState, LoaderState } from '../../state/index';
+import { EntityLoaderState } from '../../state/utils/entity-loader/entity-loader-state';
 import { NodeItem } from '../model/node-item.model';
 import { Page } from '../model/page.model';
-import { EntityLoaderState } from '../../state/utils/entity-loader/entity-loader-state';
 
 export const CMS_FEATURE = 'cms';
 export const NAVIGATION_DETAIL_ENTITY = '[Cms] Navigation Entity';
+// TODO:#4603 - mention in TODO for 2.0
 export const COMPONENT_ENTITY = '[Cms[ Component Entity';
 
 export interface StateWithCms {
   [CMS_FEATURE]: CmsState;
 }
 
 export type ComponentState = EntityLoaderState<any>;
+export type ComponentsState = EntityState<ComponentsContext>;
+
+export interface ComponentsContext {
+  // TODO:#4603 - try to use `CmsComponent` type
+  component: any;
+  pageContext: {
```

Comment:
For future reference, it would be nice to comment here *why* we slice informations per `pageContext` and why it's boolean and not the whole component state.
I guess it's because components of given UID are always the same, but may (or may not apprear) on some pages - which is reflected by the boolean stored by the `pageContext`